fetchmap/index.ts

```diff
@@ -0,0 +1,265 @@
+          finalInitialViewState = {
+            longitude: (mapData as any).mapOptions.viewState.longitude,
+            latitude: (mapData as any).mapOptions.viewState.latitude,
+            zoom: (mapData as any).mapOptions.viewState.zoom,
+            pitch: (mapData as any).mapOptions.viewState.pitch || 0,
+            bearing: (mapData as any).mapOptions.viewState.bearing || 0
+          };
+        }
+      }
+    });
+
+    currentMapData = {
+      title: mapTitles.length > 1 ? mapTitles.join(' & ') : mapTitles[0] || 'Untitled Map',
+      layers: aggregatedLayers,
+      popupSettings:
+        Object.keys(mergedPopupSettingsLayers).length > 0
+          ? {layers: mergedPopupSettingsLayers}
+          : null,
+      initialViewState: finalInitialViewState
+    };
+
+    // Update widgets
+    if (mapNameEl) {
+      mapNameEl.innerHTML = currentMapData.title;
+    }
+    if (layerCountEl) {
+      layerCountEl.innerHTML = currentMapData.layers.length.toString();
+    }
+
+    // Create and append the new legend for combined layers
+    if (currentMapData.layers && currentMapData.layers.length > 0) {
+      const legendElement = createLegend(currentMapData.layers);
+      document.body.appendChild(legendElement);
+
+      // Add event listener to toggle layer visibility
+      legendElement.addEventListener('togglelayervisibility', (event: Event) => {
+        const customEvent = event as CustomEvent<{layerId: string; visible: boolean}>;
+        const {layerId, visible} = customEvent.detail;
+        const currentDeckLayers = (deck.props.layers || []) as Layer[];
+        const newLayers = currentDeckLayers.map((layer: Layer) => {
+          if (layer && layer.id === layerId) {
+            return layer.clone({visible});
+          }
+          return layer;
+        });
+        deck.setProps({layers: newLayers});
+      });
+    }
+
+    // Finally, update the deck.gl view state and layers
+    deck.setProps({
+      initialViewState: finalInitialViewState,
+      layers: LayerFactory(currentMapData.layers)
+    });
+
+    map.jumpTo({
+      center: [finalInitialViewState.longitude, finalInitialViewState.latitude],
+      zoom: finalInitialViewState.zoom,
+      bearing: finalInitialViewState.bearing,
+      pitch: finalInitialViewState.pitch
+    });
+  } catch (error) {
+    if (mapNameEl) mapNameEl.innerHTML = 'Error loading map(s)';
+    if (layerCountEl) layerCountEl.innerHTML = 'Error';
+    currentMapData = null; // Clear data on error
+    deck.setProps({layers: []}); // Clear layers on error
+  }
+}
+
+// Initialize map selector and load default map(s)
+function setupMapSelector() {
+  if (!mapSelectorContainer || !loadMapsButton) {
+    // Fallback: Load the first map by default if UI elements are missing but configs exist
+    if (MAP_CONFIGS.length > 0) {
+      initialize([MAP_CONFIGS[0].id]);
+    } else {
+      initialize([]); // Show "No Map Selected" or similar
+    }
+    return;
+  }
+
```

Comment:
The HTML is static, can this happen? I would remove the `if (!mapSelectorContainer || !loadMapsButton)` to simplify I think.